include/nccl_ofi_param.h

```diff
@@ -18,6 +18,38 @@ extern "C" {
 #include "nccl_ofi_log.h"
 #include "nccl_ofi_pthread.h"
 
+#define OFI_NCCL_PARAM_UINT(name, env, default_value) \
```

Comment:
This is adding a lot of duplicate code. Instead, can we just
* Pre-C++: use casting when necessary
* Post-C++: use a template

Reply:
The goal here is to get to a point where our headers are safe for inclusion by both C++ and C frontends. Of course, we're trying to get to a template here. What do you suggest we do in the interim?

Reply:
Casting? Like `uint64_t unsigned_thing = (uint64)ofi_nccl_unsigned_thing()`? Or does that not work?

Reply:
If every single usage would need a cast, why not just declare it as the right type to begin with?

Reply:
Because then we have all this duplicate code, at least until we fully move over to C++.

Another idea is a helper macro, like `#define OFI_NCCL_PARAM_UINT(<args>) OFI_NCCL_PARAM_INT_GENERIC(uint64_t, <args>)` and `#define OFI_NCCL_PARAM_INT(<args>) OFI_NCCL_PARAM_INT_GENERIC(int64_t, <args>)` etc.